### PR TITLE
Fix noUncheckedIndexedAccess errors for files in the header bidding folder

### PR DESCRIPTION
**Pull request by emma-imber**
## What does this change?
Fixes noUncheckedIndexedAccess errors for files in the `header-bidding` folder

## Why?
Reduces the number of these errors by 16.

### CODE REVIEW

File: src/lib/header-bidding/prebid/bid-config.spec.ts [hunk 1]
```diff
@@ -540,11 +540,13 @@ describe('bids', () => {
...
-			delDomain: 'guardian-d.openx.net',
-			unit: '540279541',
-		});
+		if (typeof openXBid !== 'undefined') {
+			expect(openXBid.params).toEqual({
+				customParams: 'someAppNexusTargetingObject',
+				delDomain: 'guardian-d.openx.net',
+				unit: '540279541',
+			});
+		}
```
Comment (chrislomaxjones) [inline src/lib/header-bidding/prebid/bid-config.spec.ts:549]: You could make this a touch more concise with optional chaining

```suggestion
	expect(openXBid?.params).toEqual({
		customParams: 'someAppNexusTargetingObject',
		delDomain: 'guardian-d.openx.net',
		unit: '540279541',
	});
```

**Review by domlander — APPROVED** [review]
🧹

File: src/lib/header-bidding/prebid/bid-config.spec.ts (hunk 1 above)
Comment (emma-imber) [reply src/lib/header-bidding/prebid/bid-config.spec.ts:549]: Yeah fair point! I initially tried this but it triggered the `no-unnecessary-condition` rule, and I preferred to minimise the number of times I had to disable eslint rules. But still did it anyway in another file 😅

Comment (chrislomaxjones) [reply src/lib/header-bidding/prebid/bid-config.spec.ts:549]: Yeah introducing `no-unnecessary-condition` errors once we turn the rule off is a tricky one actually

**Pull request merged**